test/test_datapipe.py

```diff
@@ -248,6 +256,134 @@ def test_groupbykey_iterable_datapipe(self):
         self.assertEqual(count, 8)
 
 
+class FileLoggerSimpleHTTPRequestHandler(http.server.SimpleHTTPRequestHandler):
+    def __init__(self, *args, logfile=None, **kwargs):
+        self.__loggerHandle = None
+        if logfile is not None:
+            self.__loggerHandle = open(logfile, 'a+')
+        super().__init__(*args, **kwargs)
+
+    def log_message(self, format, *args):
+        if self.__loggerHandle is not None:
+            self.__loggerHandle.write("%s - - [%s] %s\n" %
+                                      (self.address_string(),
+                                       self.log_date_time_string(),
+                                       format % args))
+        return
+
+
+def setUpLocalServerInThread():
+    try:
+        Handler = partial(FileLoggerSimpleHTTPRequestHandler, logfile=None)
+        socketserver.TCPServer.allow_reuse_address = True
+
+        server = socketserver.TCPServer(("", 0), Handler)
+        server_addr = "{host}:{port}".format(host=server.server_address[0], port=server.server_address[1])
+        server_thread = threading.Thread(target=server.serve_forever)
+        server_thread.start()
+
+        # Wait a bit for the server to come up
+        time.sleep(3)
+
+        return (server_thread, server_addr, server)
+    except Exception:
+        raise
+
+
+def create_temp_files_for_serving(tmp_dir, file_count, file_size,
+                                  file_url_template):
+    furl_local_file = os.path.join(tmp_dir, "urls_list")
+    with open(furl_local_file, 'w') as fsum:
+        for i in range(0, file_count):
+            f = os.path.join(tmp_dir, "webfile_test_{num}.data".format(num=i))
+
+            write_chunk = 1024 * 1024 * 16
+            rmn_size = file_size
+            while rmn_size > 0:
+                with open(f, 'ab+') as fout:
+                    fout.write(os.urandom(min(rmn_size, write_chunk)))
+                rmn_size = rmn_size - min(rmn_size, write_chunk)
+
+            fsum.write(file_url_template.format(num=i))
+
+
+class TestIterableDataPipeHttp(TestCase):
+    __server_thread: threading.Thread
+    __server_addr: str
+    __server: socketserver.TCPServer
+
+    @classmethod
+    def setUpClass(cls):
+        try:
+            (cls.__server_thread, cls.__server_addr,
+             cls.__server) = setUpLocalServerInThread()
+        except Exception as e:
+            warnings.warn("TestIterableDataPipeHttp could\
+                          not set up due to {0}".format(str(e)))
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            cls.__server.shutdown()
+            cls.__server_thread.join(timeout=15)
+        except Exception as e:
+            warnings.warn("TestIterableDataPipeHttp could\
+                           not tear down (clean up temp directory or terminate\
+                           local server) due to {0}".format(str(e)))
+
+    def http_test_base(self, test_file_size, test_file_count, timeout=None,
```

Comment:
BTW, if it's a helper function, please add an underscore.

Reply:
agree, fixed